### DISCUSSION TITLE
Draw polygon programatically

### DISCUSSION THREAD

**Discussion by Asafyesh1** [geoman-io/leaflet-geoman#997] [Questions & Help] [answered]
Hello!
is there way to draw polygon on code? i have this data and i want to draw polygon when clicking on some button:
[{"Latitude":31.74627025070747,"Longitude":34.977035522460945},{"Latitude":31.804059692799363,"Longitude":34.97497558593751},{"Latitude":31.828565514766165,"Longitude":35.130844116210945},{"Latitude":31.705387248277066,"Longitude":35.13496398925782}]

Thanks,
Asaf Y

**Comment by Falke-Design** [comment, marked as answer]
This has nothing to do with Geoman. But I will give you some help:

```
var data = [{"Latitude":31.74627025070747,"Longitude":34.977035522460945},{"Latitude":31.804059692799363,"Longitude":34.97497558593751},{"Latitude":31.828565514766165,"Longitude":35.130844116210945},{"Latitude":31.705387248277066,"Longitude":35.13496398925782}];

var poly = undefined;

// loop through the data
data.forEach((d)=>{
   if(!poly){
     poly = L.polygon([L.latLng(d.Latitude, d.Longitude)]).addTo(map); // create polygon
   } else{
     poly.addLatLng(L.latLng(d.Latitude, d.Longitude)); // add latlngs to the polygon
   }
});
```
